test/solidity/Periphery/LiFiDEXAggregator.t.sol

```diff
@@ -0,0 +1,320 @@
+                fee: 3000, // - NOT USED!
+                direction: 1,
+                callback: false
+            })
+        );
+        vm.stopPrank();
+    }
+
+    function test_CanSwapViaVelodromeV2_Stable() public {
+        vm.startPrank(USER_SENDER);
+        _testSwap(
+            SwapTestParams({
+                from: USER_SENDER,
+                to: USER_SENDER,
+                tokenIn: address(USDC_TOKEN),
+                amountIn: 1_000 * 1e6,
+                tokenOut: address(USDC_E_TOKEN),
+                stable: true, // - NOT USED!
+                fee: 500, // - NOT USED!
+                direction: 0,
+                callback: false
+            })
+        );
+        vm.stopPrank();
+    }
+
+    function test_CanSwapViaVelodromeV2_Stable_Reverse() public {
+        // first perform the forward stable swap.
+        test_CanSwapViaVelodromeV2_Stable();
+
+        vm.startPrank(USER_SENDER);
+
+        _testSwap(
+            SwapTestParams({
+                from: USER_SENDER,
+                to: USER_SENDER,
+                tokenIn: address(USDC_E_TOKEN),
+                amountIn: 500 * 1e6,
+                tokenOut: address(USDC_TOKEN),
+                stable: true, // - NOT USED!
+                fee: 500, // - NOT USED!
+                direction: 1,
+                callback: false
+            })
+        );
+        vm.stopPrank();
+    }
+
+    function test_CanSwapViaVelodromeV2_FromDexAggregator() public {
+        // fund dex aggregator contract so that the contract holds USDC
+        deal(address(USDC_TOKEN), address(liFiDEXAggregator), 100_000 * 1e6);
+
+        vm.startPrank(USER_SENDER);
+        _testSwap(
+            SwapTestParams({
+                from: address(liFiDEXAggregator),
+                to: address(USER_SENDER),
+                tokenIn: address(USDC_TOKEN),
+                amountIn: USDC_TOKEN.balanceOf(address(liFiDEXAggregator)) - 1, // has to be current dex aggregator balance - 1
+                tokenOut: address(USDC_E_TOKEN),
+                stable: true, // - NOT USED!
+                fee: 500, // - NOT USED!
+                direction: 0,
+                callback: false
+            })
+        );
+        vm.stopPrank();
+    }
+
+    function test_CanSwapViaVelodromeV2_FlashloanCallback() public {
+        mockFlashloanCallbackReceiver = new MockVelodromeV2FlashLoanCallbackReceiver();
+
+        vm.startPrank(USER_SENDER);
+        _testSwap(
+            SwapTestParams({
+                from: address(USER_SENDER),
+                to: address(mockFlashloanCallbackReceiver),
+                tokenIn: address(USDC_TOKEN),
+                amountIn: 1_000 * 1e6,
+                tokenOut: address(USDC_E_TOKEN),
+                stable: true, // - NOT USED!
+                fee: 500, // - NOT USED!
+                direction: 0,
+                callback: true
+            })
+        );
+        vm.stopPrank();
+    }
+
+    struct SwapTestParams {
```

Comment:
suggest to move to the top of the contract (before it's being used)

Reply:
updated